x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/ssl/SSLReloadDuringStartupIntegTests.java

```diff
@@ -59,66 +81,121 @@ protected boolean transportSSLEnabled() {
+                waitUntilFixKeystoreIsReadyToBegin(beforeKeystoreFix); // SYNC: Cert update & ES restart
+                return super.onNodeStopped(nodeName); // ASSUME: RestartCallback will do ES start next
             }
         });
-        writtenLatch.await();
-        ensureClusterSizeConsistency();
-        ensureFullyConnectedCluster();
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Verifying the cluster...")) {
+            afterKeystoreFix.await(); // SYNC: Verify cluster after cert update
+        }
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Ensure cluster size consistency took {}ms.")) {
+            ensureClusterSizeConsistency();
+        }
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Ensure fully connected cluster took {}ms.")) {
+            ensureFullyConnectedCluster();
+        }
+    }
+
+    private void waitUntilNodeStartupIsReadyToBegin(final CountDownLatch beforeKeystoreFix) {
+        beforeKeystoreFix.countDown(); // SYNC: Cert update & ES restart
+        try {
+            final long sleepMillis = randomLongBetween(1L, 2000L); // intended sleepMillis
+            try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Sleeping " + sleepMillis + "ms before fixing...")) {
+                beforeKeystoreFix.await(); // SYNC: Cert update & ES restart
+            }
+            try (Timed t = new Timed(LOGGER, Level.INFO, "Slept {}ms, intended " + sleepMillis + "ms. Fixing can start now...")) {
+                Thread.sleep(sleepMillis); // Simulate cert update delay relative to ES start
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private void waitUntilFixKeystoreIsReadyToBegin(final CountDownLatch beforeKeystoreFix) {
+        beforeKeystoreFix.countDown(); // SYNC: Cert update & ES restart
+        try (Timed t = new Timed(LOGGER, Level.INFO, "Awaited {}ms. Node can start now...")) {
+            beforeKeystoreFix.await(); // SYNC: Cert update & ES restart
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
     }
 
-    private void copyAndAtomicMoveIfPossible(Path source, Path target) throws IOException {
+    /**
+     * Copy a source file to a destination file, overwriting if necessary.
+     * Use an intermediate temporary file, to attempt an atomic move.
+     * If atomic move fails, fall back to non-atomic move.
+     * @param source File to be copied.
+     * @param target File to be created or overwritten.
+     * @throws IOException Cannot create temp file, or copy source file to temp file, or non-atomic move temp file to target file.
+     */
+    private void atomicCopyIfPossible(Path source, Path target) throws IOException {
+        LOGGER.info("Copying [{}] to [{}]", source, target);
         Path tmp = createTempFile();
+        // LOGGER.info("Temporary file [{}] created", tmp);
```

Comment:
can you remove or re-enable the code as comments line here and below ?

Reply:
Done